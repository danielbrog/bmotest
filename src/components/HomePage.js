import React from "react";
import SearchForm from "./SearchForm";

const HomePage = () => (
  <div>
    <div className="content">HomePage</div>
    <SearchForm />
  </div>
);

export default HomePage;
