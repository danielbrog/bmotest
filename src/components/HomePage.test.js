import React from "react";
import HomePage from "./HomePage";
import ShallowRenderer from "react-test-renderer/shallow";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import mockStore from "../mocks/mockStore.json";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });
import { mount } from "enzyme";

import { axe, toHaveNoViolations } from "jest-axe";
expect.extend(toHaveNoViolations);

const configStore = configureStore([]);

describe("Homepage tests", () => {
  const store = configStore(mockStore);

  test("HomePage renders with a title", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HomePage />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it("should pass accessibility tests", async () => {
    const wrapper = mount(
      <Provider store={store}>
        <HomePage />
      </Provider>
    );
    const results = await axe(wrapper.getDOMNode());
    expect(results).toHaveNoViolations();
  });
});
