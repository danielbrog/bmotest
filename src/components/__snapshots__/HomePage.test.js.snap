// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage tests HomePage renders with a title 1`] = `
<div
  className="content"
>
  <div
    className="main__title"
    role="banner"
  >
    Find Restaurants in your city!
  </div>
  <Memo(Connect(SearchForm)) />
  <Memo(Connect(RestaurantList)) />
</div>
`;
